### ISSUE TITLE
MQTT-Client BP-Node "Connect" Event-Dispatcher "OnConnect" does not fire.

### ISSUE THREAD

**Issue by brugr9** [brugr9/Heartbeat51#1]
Steps to Reproduce:

## A. As Is

1. Startup Mosquitto Brocker
2. Startup UE Project "Heartbeat51", Launch PIE
3. MQTT-Client BP-Node "Connect" works, see Fig. Wireshark dissection
4. MQTT-Client BP-Node "Connect" Event-Dispatcher "OnConnect" does not fire, see Fig. 

![Screenshot-Wireshark-1883-connect](https://user-images.githubusercontent.com/1733987/212481319-6648941f-6f95-4e61-ad6d-6b7dc35bf56b.png)
*Fig.: Wireshark Dissecting MQTT Connect Command*

![Screenshot 2023-01-14 164539](https://user-images.githubusercontent.com/1733987/212481358-83595ced-94b3-4ce4-8ee8-2288fc4215fb.png)
*Fig.: MQTT-Client BP-Node "Connect" Event-Dispatcher "OnConnect" does not fire*

## B. Workaround

1. Startup Mosquitto Brocker
2. Startup UE Project "Heartbeat51"
    * To enable the subscription, wire the "Connect" and "Subscribe" nodes temporarily, see Fig.
    * Launch PIE
3. Startup MQTT Explorer
    * connect to mqtt://127.0.0.1:1883
    * publish a message using topic psl/hr
4. The Editor crashes

![Screenshot 2023-01-14 164728](https://user-images.githubusercontent.com/1733987/212481509-8836a6fb-030c-488c-9086-3e51f6a1977f.png)
*Fig.: "Connect" and "Subscribe" nodes temporarily wired*

![MqttExplorer-0](https://user-images.githubusercontent.com/1733987/212479684-500e72b7-6074-42b3-876e-bb466eb44c21.png)
*Fig.: MQTT-Explorer Connection*

![MqttExplorer-Publish](https://user-images.githubusercontent.com/1733987/212479685-35faad93-eed9-4100-bd02-2619875adeb3.png)
*Fig.: MQTT-Explorer Publish*

Listing: Topic psl/hr, example Payload in JSON
```json
{
  "clientId": "MyPSL-01",
  "deviceId": "12345678",
  "sessionId": 1234567890,
  "timeStamp": 1234567890123,
  "hr": 64,
  "rr": [
    833
  ]
}
```

![Wireshark-Publish](https://user-images.githubusercontent.com/1733987/212479688-1963b3fa-3a3a-4950-8a17-c63374774a91.png)
*Fig.: Wireshark Dissecting MQTT Explorer Publication*

---

* See also snippet "MQTT Plugin" by DrYannick1: https://dev.epicgames.com/community/snippets/EoX/unreal-engine-mqtt-plugin; says:  *"Event 'OnConnect' is not working. Some received payloads make the engine crash."*
* reported bug to https://www.unrealengine.com/en-US/support/report-a-bug
* https://issues.unrealengine.com/issue/search?q=MQTT

**Comment by julius-gummersbach** [comment]
Can confirm this is still an issue. If there are any updates let me know. I have been trying to get mqtt working with Unreal Engine for days.

**Comment by julius-gummersbach** [comment]
> https://issues.unrealengine.com/issue/search?q=MQTT

Your Issue seems to not be there anymore btw
![grafik](https://github.com/brugr9/Heartbeat51/assets/97508212/bf159034-53b3-4292-adbb-bbeb708309a5)

**Comment by brugr9** [comment]
Thanks Julius. Yes, I know - the issue has not been listed so far ... does not seem to have been judged as an important issue. The MQTT plugin is beta btw.

**Comment by julius-gummersbach** [comment]
Yes, I gave up on mqtt and now use the HTTP Plugin to send requests to a REST Api on my computer which returns the last mqtt messages. works fine for my  usecase but far from ideal...

**Comment by HomeoStasis-0** [comment]
Hi Roland, I am currently trying to implement this into a project, but I am using the verity sense sensor instead of the chest strap. Would you happen to know what differences there would be in the unreal code. Unlike the chest strap using the ECG, I am only needing the HR but this is my first time using unreal.  Any help would be great!
